libs/vault/src/cipher-form/components/custom-fields/custom-fields.component.ts

```
@@ -186,6 +197,11 @@ export class CustomFieldsComponent implements OnInit, AfterViewInit {
       this.isPartialEdit = true;
       this.customFieldsForm.disable();
     }
+
+    const collectionId = this.route.snapshot.queryParams?.collectionId;
+    this.disallowHiddenField = this.cipherFormContainer.config.collections.some(
```

❓ Why do we need this piece? I would expect the `canEdit(type)` helper you added above could be used to determine if the user could modify hidden fields. 

In case of _new_  ciphers, we shouldn't need to restrict them from creating hidden fields initially (we allow them to set a password, even though they'll immediately lose "access" to it after saving to a collection with edit, except password)

---

@shane-melton I agree this is a great approach. Unfortunately there's an issue where, after initial creation, the form config doesn't populate the `originalCipher` so we cannot establish permission. If the user clicks "edit" without closing the form then they will again see the "Hidden" option but any additional hidden fields will be rejected. If they close out the form and re-enter the will no longer see the "Hidden" option. Is this acceptable?

---

Hmm, the `originalCipher` should be populated on save [here](https://github.com/bitwarden/clients/blob/41e1d4e7bfe22caabd780c4caf8033e8131fb4e1/apps/web/src/app/vault/components/vault-item-dialog/vault-item-dialog.component.ts#L389-L390) (at least on Web). 

With that said, I see the endpoint returns the[ provided cipher model back to the user in the response](https://github.com/bitwarden/server/blob/abe593d22122cda682b7b789a4eadb6ceccfad85/src/Api/Vault/Controllers/CiphersController.cs#L180-L185) instead of fetching the saved cipher with the permissions as defined by the server. If we instead tweak the server to instead return `Get(cipher.Id)` we will get back the correct `viewPassword` value and the dialog should behave as expected.

---

Also, the collectionId query param will not work if the user is not filtering by a collection when they open the dialog. We'd need to see which collections the original cipher belongs to and check that way (but ideally we'd just let the server do that for us instead as described above)